### ISSUE TITLE
[DISCUSSION] Naming of serializers and viewsets

### ISSUE THREAD

**Issue by lemontree210** [samanthasgroup/django-webapps#177]
We have two clients for our views:
- Coordinator's dashboard (ToolJet)
- Bot

We also have views that can be used both by coordinator and by bot (e.g. #174).

In the future we cannot exclude creation of a teacher's dashboard (not in MVP) and admin's dashboard (right now we plan to implement basic functionality for the admins in Django admin site) and views for them. Unfortunately, the free version of ToolJet doesn't support different roles, so right now Tooljet will only provide a dashboard for coordinators. 

I know that the dashboard category was initially called "public", but this was no final decision: we decided that we would take it up later when need arises. 

The objections against "public" are:

- In a way, none of the views are really public because the system will be closed for the outside world
- If more dashboards get created, "public" gets even more confusing.

So I suggest that, without resorting to "brand" names like "ToolJet", we can be sure that we have some bot (or bots) and some dashboard that access data from the backend. So the serializers could have names like `BotTeacherSerializer`, `CoordinatorDashboardTeacherSerializer` (same for viewsets). For the cases when both clients can use the same serializer/viewset, we can create some `CommonTeacherSerializer` or just `TeacherSerializer`.

**Comment by lemontree210** [comment]
Let's discuss @Auch-Auch @ivanBarinsky @paulin-mipt  @Roxe322

**Comment by Roxe322** [comment]
In my opinion, it's totally fine to have multiple views doing same stuff, but in different routers. 
For example, you can start group both in `/public/group/start/` and `/group/start/`. 
These router prefixes are just to show that something is used by bot / by dashboard / by anything else (we could add any other prefixes in future)

**Comment by Roxe322** [comment]
What do you suggest now? rename all `public` stuff to `coordinator_dashboard`? 
For me `public` is everything for end user (despite permissions, authorization, etc.)  and `non-public` is for internal purposes (bot, API for other apps internally developed apps)

**Comment by Auch-Auch** [comment]
For me public is fine but I am not entirely sure whether we actually need it. I usually decide this way: If I have users that need to use API without any authentication I use `public` router otherwise I explicitly define what views a particular user (authorized) can use (user can be admin, bot or whatever). So for me this `public` seems redundant, but I probably do not fully understand business logic

**Comment by lemontree210** [comment]
Hm @Roxe322 if you think of it, none of the views are actually public: coordinators won't even know that an API exists and everything is happening "under the hood". So is there actually any difference to the way the user interacts with the API via the Bot. None of the API views we are writing are intended for end users.

@Auch-Auch I have the same intuition, that `public` is everything that requires no authentication, but I see the logic in @Roxe322 's words too. If we have a public API, the user might still be required to pass some sort of token. But disregarding that, the end-users don't interact with the API directly anyway. So it's not really public in that sense either.

To top this off, for some reason we now have `serializers.teacher.public` module containing classes `Teacher[Read/Write]Serializer` and in `serializers.group.internal` we have `Group[Read/Write]Serializer` that @Auch-Auch apparently named after the teacher serializers, but they are obviously "public" in @Roxe322 's sense.

Basically what I'm saying is that with this logic spinning out of control and none of the serializers actually being public, maybe we should unify these `internal` and `public` modules where they exist. We could leave `minified.py` in place, I think it makes sense.

As for viewsets, I think that `Public` can be replaced with `Dashboard` where it makes sense to keep a separate class.

**Comment by Roxe322** [comment]
I think we have to detach "dashboard" views and "bot" views (maybe any others, but for now we have only two "clients" - bot and dashboard) in modules hierarchy somehow. 
So my idea was to call it public (for dashboard - since it's used by "end users" (I understand that it's used by tooljet and so on)) and "internal" (since it's used by bot - not "end users").
It's totally fine to change naming to something less confusing (e.g Public -> Dashboard ; Internal -> Bot), but please let's keep them detached

**Comment by lemontree210** [comment]
OK, I agree. @Auch-Auch what do you think?

**Comment by Auch-Auch** [comment]
Yes, that would make sense. It is still a little bit confusing for me to completely split it when we have some overlapping endpoints, and we can not reuse them but with DRF it might make sane because it is relatively easy to define new views. In my practice, I usually try to create more or less generic API but it has its own downsides that might be not relevant here.

**Comment by lemontree210** [comment]
Great. @Auch-Auch if you find it an interesting task, you can make a PR. It's totally OK if you'd rather prefer something else (might not be the most interesting thing to do), in which case I'll make a PR (since I encouraged this discussion 😃). Just let me know.

**Comment by Auch-Auch** [comment]
Yep, why not, assign it to me, but the PR itself will probably be at weekends
